src/expressions/operators/compares/generalCompare.ts

```diff
@@ -108,40 +109,63 @@ export default function generalCompare(
 	);
 }
 
-export function generatePrefabFunction(
+export function getGeneralCompareEvaluationFunction(
 	operator: string,
-	firstType: ValueType,
-	secondType: ValueType
+	firstType: SequenceType,
+	secondType: SequenceType
 ): (first: ISequence, second: ISequence, dynamicContex: DynamicContext) => boolean {
 	operator = OPERATOR_TRANSLATION[operator];
 
 	const [firstTargetType, secondTargetType]: [ValueType, ValueType] = typeCheck(
```

Comment:
typeCheck could probably get a more descriptive name. Something along the lines of 'determineGeneralCompareTypes'